pygen/pygen_src/isa/riscv_instr.py

```diff
@@ -0,0 +1,341 @@
+from collections import defaultdict
+from pygen_src.riscv_instr_gen_config import *
+from pygen_src.riscv_instr_pkg import *
+from pygen_src.isa import rv32i_instr
+import random
+from bitstring import BitArray, BitStream
+import logging
+from copy import copy
+class riscv_instr:
+    instr_registry = {}
+    def __init__(self): 
+        self.instr_names = []
+        self.instr_name = None
+        self.instr_group =  defaultdict(list) 
+        self.instr_category = defaultdict(list)
+        self.basic_instr = []
+        self.instr_template = {}
+        
+        self.exclude_reg = []
+        self.include_reg = []
+        
+        self.imm_len = 0
+        self.csr = None
+        self.rs2 = None
+        self.rs1 = None
+        self.rd = None
+        self.imm = BitArray(uint = 0 , length = 32)
+        self.idx = -1
+        self.has_rs1 = 1
+        self.has_rs2 = 1
+        self.has_rd = 1
+        self.has_imm = 1
+
+        self.imm_mask = BitArray(uint = 4294967295 , length = 32)
+        self.is_branch_target = None
+        self.has_label  = 1
+        self.atomic = 0
+        self.branch_assigned = None
+        self.process_load_store = 1
+        self.is_compressed = None
+        self.is_illegal_instr = None
+        self.is_hint_instr = None
+        self.is_floating_point = None
+        self.imm_str = None
+        self.comment = None
+        self.label = None
+        self.is_local_numeric_label = None
+        
+        #Fields Added for debugging These fields are actually from a different files. 
+        self.unsupported_instr = []
+        self.XLEN = 32
+        self.supported_isa  = [riscv_instr_group_t.RV32I]
+        self.implemented_csr = [] 
+    @classmethod
+    def register (cls, instr_name):
+        print("Registering {}".format(instr_name.name))
+        cls.instr_registry[instr_name.name] = 1
+        if instr_name == None:
+            print("\n")
+        return 1
+
+    def create_instr_list(self, cfg):
+        self.instr_names.clear()
+        self.instr_group.clear()
+        self.instr_category.clear()
+        for instr_name, values in self.instr_registry.items():
+
+            if(instr_name in self.unsupported_instr):
+                continue
+            instr_inst = self.create_instr(instr_name) 
+            self.instr_template[instr_name] = instr_inst
+
+            if (not instr_inst.is_supported(cfg)):
+                continue
+            if ((self.XLEN != 32) and (instr_name == "C_JAL")):
+                continue
+            if (("SP" in cfg.reserved_regs) and (instr_name == "C_ADDI16SP")):
+                continue
+            if (cfg.enable_sfence and instr_name == "SFENCE_VMA"):
+                continue
+            if (instr_name in ["FENCE","FENCE_I","SFENCE_VMA"]):
+                continue
+            if (instr_inst.group in self.supported_isa):
```

Comment:
There should be some more conditional statements here inside the `if (...)`, see https://github.com/google/riscv-dv/blob/master/src/isa/riscv_instr.sv#L124

Reply:
Fixed it!